src/main/java/org/jenkinsci/plugins/testrail/TestRailNotifier.java

```diff
@@ -191,27 +196,43 @@ public Results addSuite(Testsuite suite, String parentId, ExistingTestCases exis
 
         if (suite.hasCases()) {
             for (Testcase testcase : suite.getCases()) {
-                int caseId;
+                int caseId = 0;
+                boolean addResult = false;
                 try {
                     caseId = existingCases.getCaseId(suite.getName(), testcase.getName());
+                    addResult = true;
                 } catch (ElementNotFoundException e) {
-                    caseId = existingCases.addCase(testcase, sectionId);
-                }
-                CaseStatus caseStatus;
-                Float caseTime = testcase.getTime();
-                String caseComment = null;
-                Failure caseFailure = testcase.getFailure();
-                if (caseFailure != null) {
-                    caseStatus = CaseStatus.FAILED;
-                    caseComment = (caseFailure.getMessage() == null) ? caseFailure.getText() : caseFailure.getMessage() + "\n" + caseFailure.getText();
-                } else if (testcase.getSkipped() != null) {
-                    caseStatus = CaseStatus.UNTESTED;
-                } else {
-                    caseStatus = CaseStatus.PASSED;
+                    if (this.createNewTestcases) {
+                        caseId = existingCases.addCase(testcase, sectionId);
+                        addResult = true;
+                    }
                 }
-
-                if (caseStatus != CaseStatus.UNTESTED){
-                    results.addResult(new Result(caseId, caseStatus, caseComment, caseTime));
+                if (addResult) {
+                    int caseStatus;
```

Comment:
It looks like both code blocks were retained when you merged from upstream. You'll want to delete everything from this line to line 221.